src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.NameGenerator.cs

```diff
@@ -57,6 +57,15 @@ private static ImmutableArray<Words> GetInterleavedPatterns(ArrayBuilder<TextSpa
                     result.Add(GetLongestBackwardSubsequence(length, breaks, baseName, pluralize));
                 }
 
+                for (var index = result.Count - 1; index >= 0; index--)
+                {
+                    if (char.IsDigit(result[index][0], 0))
```

Comment:
is this check correct?  a few thingsthat worry me:

1. how do we know that `result[index]` is non-empty?  Indexing into `[0]` might be unsafe.
2. we're only checking the first word in `Words`, why is that correct?  What about the other words in it?

Feels like we need to check *every* word to see if it starts with a digit.

Reply:
1. The strings inside the Words struct were created by StringBreaker.GetWordParts(). StringBreaker will only add TextSpans that are not emtpy.
2. Every Words-struct is used by the DeclarationNameCompletionProvider to create exactly one name suggestion, based on current NamingStyle. Only the first word will be the start of the generated name.

Reply:
1. is the above true given that some of the strings added were also created by GetLongestBackwardSubsequence and so on?
2. if what you say is correct, *great*.  but that must be documented in the code.  please add a comment explaining as much.  thanks! :)

Reply:
1. Yes, the GetLongestBackwardSubsequence and GetLongestForwardSubsequence will create subsequences containing a subset of the strings that were originally create by the StringBreaker. No new texts will be created from scratch.
2. Done. Please let me know if you think the new comment is helpful.